### DISCUSSION TITLE
Translation files in flat routes

### DISCUSSION THREAD

**Discussion by fyra3or** [sergiodxa/remix-i18next#123] [Ideas]
With the [v2 route file naming](https://remix.run/docs/en/1.14.0/file-conventions/route-files-v2) in Remix, we can organize our route specific components and other code closer to the routes that use them.

Having global and glossary style translations in the public folder is good. But it often become time consuming to manage route specific translations there, even with different translation files/namespaces. It would be great to be able to put the route specific translations in the route folder, together with the the other route specific code.

We could then load those translation files in the meta function in the route file (or have them loaded automatically by the library base on file name maybe..).

**Comment by sergiodxa** [comment]
You will still need them in public so the browser can download them in client-side navigations, unless you only use remix-i18next inside loaders and pass translated texts as part of the loader response.
